cost-analyzer/templates/_helpers.tpl

```diff
@@ -398,31 +403,264 @@ The implied use case is {{ template "cost-analyzer.filterEnabled" .Values }}
 {{- end -}}
 
 {{/*
-This template runs the full check for leader/follower requirements in order to determine
-whether it should be configured. This template will return true if it's enabled and all
-requirements are met.
+ Check KC 2.0 values requirements that may differ
+*/}}
+{{ if .Values.federatedETL }}
+  {{ if .Values.federatedETL.primaryCluster }}
+    {{ fail "In Kubecost 2.0, all federated configurations must be set up as secondary" }}
+  {{ end }}
+{{ end }}
+
+{{ if .Values.kubecostModel }}
+  {{ if .Values.kubecostModel.openSourceOnly }}
+    {{ fail "In Kubecost 2.0, kubecostModel.openSourceOnly is not supported" }}
+  {{ end }}
+{{ end }}
+
+{{/*
+ Aggregator config reconciliation and common config
 */}}
-{{- define "cost-analyzer.leaderFollowerEnabled" }}
-    {{- if .Values.kubecostDeployment }}
-        {{- if .Values.kubecostDeployment.leaderFollower }}
-            {{- if .Values.kubecostDeployment.leaderFollower.enabled }}
-                {{- $replicas := .Values.kubecostDeployment.replicas | default 1 }}
-                {{- if not .Values.kubecostModel.etlFileStoreEnabled }}
-                    {{- "" }}
-                {{- else if (eq (quote .Values.kubecostModel.etlBucketConfigSecret) "") }}
-                    {{- "" }}
-                {{- else if not (gt (int $replicas) 1) }}
-                    {{- ""}}
-                {{- else }}
-                    {{- "true" }}
-                {{- end }}
-            {{- else }}
-                {{- "" }}
-            {{- end }}
-        {{- else }}
-            {{- "" }}
-        {{- end }}
-    {{- else }}
-        {{- "" }}
+{{ if eq (include "aggregator.deployMethod" .) "statefulset" }}
+  {{ if .Values.kubecostAggregator }}
+    {{ if (not .values.kubecostAggregator.aggregatorDbStorage) }}
+      {{ fail "In Enterprise configuration, Aggregator DB storage is required" }}
+    {{ end }}
+  {{ end }}
+{{ end }}
+
+
+{{- define "aggregator.containerTemplate" }}
+- name: aggregator
+{{- if .Values.kubecostAggregator.containerSecurityContext }}
+  securityContext:
+    {{- toYaml .Values.kubecostAggregator.containerSecurityContext | nindent 4 }}
```

Comment:
have you tested this rendering? I know sometimes template scope gets weird, and this is probably fine, but just wanted to make sure you have looked at that

Reply:
Yep! If my memory serves, a couple of days ago I remember modifying this because the security context block (and a few others) were all messed up until I made this change.